spark/dl/src/main/scala/com/intel/analytics/bigdl/nn/Linear.scala

```diff
@@ -81,6 +81,12 @@ class Linear[T: ClassTag](
     zeroGradParameters()
   }
 
+  override def doComputeOutputShape(inputShape: Activity): Activity = {
```

Comment:
The idea is all of the old-style layers should implement this method for calc the outputshape by a given inputshape.